cli-toolchain/usage.md

```diff
@@ -696,58 +696,136 @@ Babel needs to be configured properly so that the transpiled code can be run on
+
+The `plugins.json` file is stored at path `<os.homedir>/.amplify/plugins.json`. Unless you really know what you are doing, you should NOT manually edit this file, otherwise you run the risk of corrupting your local installation of the Amplify CLI. 
+
+The `plugins.json` file will be created or updated in the following situations: 
 
+* If the `plugins.json` file is not found when the Amplify CLI Core tries to access it, the Amplify CLI Core will create this file and scan the local environment for plugins, and then store the information in the file.
+* If  the last scan time was more than one day (configurable) ago, the Amplify CLI Core will scan again and update the information. 
+* If inaccuracy is detected, e.g. a specified plugin can not be loaded, the Amplify CLI Core will scan again and update the information. 
+* After the execution of any of the `amplify plugin` commands that could change it, e.g. `amplify plugin scan`, `amplify plugin add/remove`, etc..
 
-The Amplify CLI uses <a href="https://github.com/infinitered/gluegun##readme" target="_blank">gluegun</a>. The CLI core provides the pluggable platform, and most of the CLI category functions are implemented as plugins. The CLI core searches for plugins in the global `node_modules` directory, and its own `node_modules` directory. Plugins are recognized by the `amplify-` prefix in the package names.
+By default, the CLI core searches for plugins in its parent directory, its local `node_modules` directory, and the global `node_modules` directory; and plugins are recognized by the `amplify-` prefix in the package names.
 
-Plugins communicate with the CLI core, and with each other, through the project metadata. The CLI core provides the read and write access to the project metadata for the plugins. The project metadata is stored in file `amplify/backend/amplify-meta.json` in the user project, see [below](##meta) for more details of the meta data file. 
+Plugins communicate with the CLI core, and with each other, through the project metadata. The CLI core provides the read and write access to the project metadata for the plugins. The project metadata is stored in file `amplify/backend/amplify-meta.json` in the user project. 
+
+### Plugin types
+![Image]({{media_base}}/AmplifyCliConcept.jpg)
 
-#### Plugin types
 There are four types of plugins
 - category
 - provider
 - frontend
-- general purpose
+- util
 
 #### Category plugin
-Recognized by the `amplify-category-` prefix in the package name, a category plugin wraps up the logic to create and manage one category of backend resources in the cloud. It defines the "shape" of the cloud resources based on user (the developer) input, constructs parameters to CRUD cloud resource, and exports relevant cloud resource information to the project metadata.
+Official category plugins are recognized by the `amplify-category-` prefix in the package name.<br/>
```

Comment:
What do we mean by official here? 
Do we mean a category plugin is recognized by the 'amplify-cateogry-' prefix ?
If yes, then we should remove the work official